File: wix-petri-core/src/main/java/com/wixpress/petri/laboratory/RegisteredUserInfoType.java
package com.wixpress.petri.laboratory;

import com.wixpress.petri.experiments.domain.Experiment;
import com.wixpress.petri.experiments.domain.TestGroup;
import scala.Option;
import scala.Some;

import java.util.UUID;

public class RegisteredUserInfoType implements UserInfoType {

    private final UUID userId;

    public RegisteredUserInfoType(UUID userId) {
        this.userId = userId;
    }

    public boolean isAnonymous() {
        return false;
    }

    public TestGroup drawTestGroup(Experiment exp) {
        return new GuidTestGroupAssignmentStrategy().getAssignment(exp, userId.toString());
    }

    @Override
    public Option<UUID> persistentKernel() {
        return new Some(userId);
    }

    @Override
    public boolean shouldPersist(){
       return true;
    }

}

File: wix-petri-core/src/main/java/com/wixpress/petri/laboratory/RegisteredUserInfoType.scala
package com.wixpress.petri.laboratory

import java.util.UUID

import com.wixpress.petri.experiments.domain.{TestGroup, Experiment}

case class RegisteredUserInfoType(userId: UUID) extends UserInfoType {

  def isAnonymous: Boolean = false

  def drawTestGroup(exp: Experiment): TestGroup = ConductionStrategy.drawTestGroupForId(exp, userId.toString)

  def persistentKernel: Option[UUID] = Some(userId)


}